### ISSUE TITLE
cf_jackknife is never inherited

### ISSUE THREAD

**Issue by kostrzewa** [HISKP-LQCD/hadron#202]
It seems to me that the `cf_jackknife` mixin never exists.

**Comment by kostrzewa** [comment]
Yet, `summary.cf` tests for it:

```r
  if (inherits(cf, 'cf_jackknife')) {                                                                                                             
    out <- cbind(out, jackknife.se=cf$jackknife.se)
    out <- cbind(out, jab.se=cf$jack.boot.se)

    if( has_icf(cf) ){
      out <- cbind(out, ijackknife.se = cf$ijackknife.se)
      out <- cbind(out, ijab.se = cf$ijack.boot.se)
    }
  }

```

**Comment by martin-ueding** [comment]
Yes, this is correct. We discussed whether we wanted to have jackknife and bootstrap as separate entities in the code. Most of our functions do not care about the exact nature of the samples, instead they mostly map a function over a set of samples. Sometimes they need to compute the error or covariance, but for that the `cf_boot` mixin comes with the `error_fn` and `cov_fn` fields. I believe that we said that all `boot` variables actually stand for samples now, and the actual nature of the samples is given by the sample type field.

This way the `summary.cf` should not really test for that but just use the `error_fn` to produce the error estimates.

These changes have broken the jackknife after bootstrap, I believe. But we wanted to have less work while doing the rho analysis with jackknife at that point in time.

Ideally I'd like for the functions `f` like `+.cf` to just produce a kernel that takes one correlator C and gives a correlator, like `f: C → C`. Then we have a `map` which takes a correlator with samples, `(C, {C})`, and applies it such that we have `(f C, {f C})`. For functions like the `plot.cf` where we need the errors, we use a function like `error_fn` to compute it from the data.

And instead of having a growing data structure with additional mixins I would like to have a hierarchy where we have a `cf_orig` class (not mixin), which just has the original data. Then the `cf_sampled` would have a `orig` *member* and also the samples and the type. If one wants to jackknife that again, one would again get a `cf_sampled` but this time the `orig` member is the previous `cf_sampled` instance.

**Comment by kostrzewa** [comment]
> These changes have broken the jackknife after bootstrap, I believe. But we wanted to have less work while doing the rho analysis with jackknife at that point in time.

jab should just test for `resampling_method` and so should `summary.cf`. I might just fix this in #200 

> Ideally I'd like for the functions f like +.cf to just produce a kernel that takes one correlator C and gives a correlator, like f: C → C. Then we have a map which takes a correlator with samples, (C, {C}), and applies it such that we have (f C, {f C}). For functions like the plot.cf where we need the errors, we use a function like error_fn to compute it from the data.

Sure, that would be ideal.
